drivers/atrvt100.s

```diff
@@ -1,6 +1,1709 @@
+        sta SHFLOK
+        sta INVFLG
+InitScr1              ; called from ExitScr
+        ldx #Rows-1
+IS1     txa
+        pha           ; save X
+        jsr ErLn      ; erase line
+        pla
+        tax           ; restore X
+        dex           ; previous line
+        bpl IS1
+        ; --- put crsr ---
+        ldx #$00
+        ldy #$00
+        sty COLCRS+1  ; sanity
+        jsr Plot
+
+        rts
+
+; -------------------------------------
+; Exit Screen
+;
+; -------------------------------------
+
+ExitScr
+        lda LMARGN_save
+        sta LMARGN
+        lda SHFLOK_save
+        sta SHFLOK
+        lda INVFLG_save
+        sta INVFLG
+        jmp InitScr1
+
+; *************************************
+; *
+; * ASCII and ScreenCode tables
+; *
+; *************************************
+
+; -------------------------------------
+; table line drawing to ScreenCode
+;
+; This tabel is used to convert incoming
+; line drawing chars.
+; -------------------------------------
+
+ltsc;_0  _1  _2  _3  _4  _5  _6  _7  _8  _9  _a  _b  _c  _d  _e  _f
+
+; --- unused -------------------------------------------------------
+.byt $00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00  ; 0_
+.byt $00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00  ; 1_
+
+; --- reverse ------------------------------------------------------
+;     ◆   ▒   ␉   ␌   ␍   ␊   °   ±   ␤   ␋   ┘   ┐   ┌   └   ┼   ⎺
+;    ' '  ▒  ' ' ' ' ' ' ' ' ' ' ' ' ' ' ' '  |  ' ' ' '  |   |   _
+.byt $20,$57,$20,$20,$20,$20,$20,$20,$20,$20,$7c,$20,$20,$7c,$7c,$1f  ; 2_
+;     ⎻   ─   ⎼   ⎽   ├   ┤   ┴   ┬   │   ≤   ≥   π   ≠   £   ·  ' '
+;     _   _   _   _   |   |   _   _   |  ' ' ' ' ' ' ' ' ' ' ' ' ' '
+.byt $1f,$1f,$1f,$1f,$7c,$7c,$1f,$1f,$7c,$20,$20,$20,$20,$20,$20,$20  ; 3_
+
+; --- unused -------------------------------------------------------
+.byt $00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00  ; 4_
+.byt $00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00,$00  ; 5_
+
+; --- normal -------------------------------------------------------
+;     ◆   ▒   ␉   ␌   ␍   ␊   °   ±   ␤   ␋   ┘   ┐   ┌   └   ┼   ⎺
+;     ◆   ▒  ' ' ' ' ' ' ' ' ' ' ' ' ' ' ' '  ⎸  ' '  _   ⎿   ⎿   ⎺
+;.byt $5b,$56,$a0,$a0,$a0,$a0,$a0,$a0,$a0,$a0,$5f,$a0,$9f,$54,$54,$4c  ; 6_
+.byt $5b,$56,$a0,$a0,$a0,$a0,$a0,$a0,$a0,$a0,$43,$45,$51,$5a,$54,$4c  ; 6_
+;     ⎻   ─   ⎼   ⎽   ├   ┤   ┴   ┬   │   ≤   ≥   π   ≠   £   ·  ' '
+;     ─   _   _   _   ⎿   ⎸   ⎿   _   ⎸  ' ' ' ' ' ' ' ' ' ' ' ' ' '
+;.byt $53,$9f,$9f,$9f,$54,$5f,$54,$9f,$5f,$a0,$a0,$a0,$a0,$a0,$a0,$a0  ; 7_
+.byt $53,$52,$9f,$9f,$54,$5f,$54,$9f,$7c,$a0,$a0,$a0,$a0,$a0,$a0,$a0  ; 7_
+
+; -------------------------------------
+; table keyboard to ASCII
+;
+; This table is used to prepare keyboard
+; input for sending over the serial
+; line.
+;
+; ascii = $82 means HELP key
+; ascii = $81 means ignore key
+; ascii = $80 means do something
+;             complicated (command key)
```

Comment:
Is there a special reason why you moved away from using $ff, $fe, ... downwards for special purposes? If not I'd appreciate if you would stick to the original approach used on the C64 and Apple II.